File: solutions/csharp/bird-watcher/BirdWatcher.cs
using System;

class BirdCount
{
    private int[] birdsPerDay = new int[7];
    private const int _today = 6;

    public BirdCount(int[] birdsPerDay) => this.birdsPerDay = birdsPerDay;

    public static int[] LastWeek() => new[] { 0, 2, 5, 3, 7, 8, 4 };

    public int Today() => birdsPerDay[_today];

    public void IncrementTodaysCount() => birdsPerDay[_today] += 1;

    public bool HasDayWithoutBirds() => Array.IndexOf(birdsPerDay, 0) != -1;

    public int CountForFirstDays(int numberOfDays)
    {
        int sum = 0;
        for (int i = 0; i < numberOfDays; i++)
        {
            sum += birdsPerDay[i];
        }
        return sum;
    }

    public int BusyDays()
    {
        const int busyDayBirds = 5;
        int days = 0;
        foreach (var birds in birdsPerDay)
        {
            days += birds < busyDayBirds ? 0 : 1;
        }
        return days;
    }
}

File: solutions/java/bird-watcher/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;
    private final int[] birdsLastWeek = { 0, 2, 5, 3, 7, 8, 4 };
    private final int birdsPerBusyDay = 5;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return birdsLastWeek;
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1] += 1;
    }

    public boolean hasDayWithoutBirds() {
        for (int birdsSeen : birdsPerDay) {
            if (birdsSeen == 0)
                return true;
        }
        return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int totalBirds = 0;
        for (int i = 0; i < Math.min(numberOfDays, birdsPerDay.length); i++) {
            totalBirds += birdsPerDay[i];
        }
        return totalBirds;
    }

    public int getBusyDays() {
        int busyDays = 0;
        for (int birdsSeen : birdsPerDay) {
            if (birdsSeen >= birdsPerBusyDay)
                busyDays += 1;
        }
        return busyDays;
    }
}
